app/actions/manageSafetyDepositBoxActions.js

```diff
@@ -212,7 +214,6 @@ export function deleteVaultPath(path, token) {
         })
             .then((response) => {
                 log.info("SUCCESS", response)
-                location.reload(); // todo hack until I get back from vacation
```

Comment:
You need to dispatch a new event here that will remove the newly deleted key from the state.

I would imaging the reducer for that to look something like the following

    [action.REMOVE_FETCHED_VAULT_SECRET]: (state, payload) => {
        let existingMap = state.vaultSecretsData
        let newMap = {}
        let deletedKey = payload

        for (let key in existingMap) {
            if (existingMap.hasOwnProperty(key) && key != deletedKey) {
                newMap[key] = existingMap[key]
            }
        }

        return Object.assign({}, state, {
            vaultSecretsData: newMap
        })
    }

As of right now, when you delete a Vault node the old data will stay on the page.